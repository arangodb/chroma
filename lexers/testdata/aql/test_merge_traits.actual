FOR c IN Characters
  RETURN MERGE(c, {
    traits: (
      FOR key IN c.traits
        FOR t IN Traits
          FILTER t._key == key
          RETURN t.en
    )
  })