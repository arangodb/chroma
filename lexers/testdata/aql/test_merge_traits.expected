[
  {"type":"KeywordReserved","value":"FOR"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"c"},
  {"type":"Text","value":" "},
  {"type":"KeywordReserved","value":"IN"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"Characters"},
  {"type":"Text","value":"\n  "},
  {"type":"KeywordReserved","value":"RETURN"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"MERGE"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"c"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"Name","value":"traits"},
  {"type":"Operator","value":":"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":"\n      "},
  {"type":"KeywordReserved","value":"FOR"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"key"},
  {"type":"Text","value":" "},
  {"type":"KeywordReserved","value":"IN"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"c"},
  {"type":"Punctuation","value":"."},
  {"type":"Name","value":"traits"},
  {"type":"Text","value":"\n        "},
  {"type":"KeywordReserved","value":"FOR"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"t"},
  {"type":"Text","value":" "},
  {"type":"KeywordReserved","value":"IN"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"Traits"},
  {"type":"Text","value":"\n          "},
  {"type":"KeywordReserved","value":"FILTER"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"t"},
  {"type":"Punctuation","value":"."},
  {"type":"Name","value":"_key"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"key"},
  {"type":"Text","value":"\n          "},
  {"type":"KeywordReserved","value":"RETURN"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"t"},
  {"type":"Punctuation","value":"."},
  {"type":"Name","value":"en"},
  {"type":"Text","value":"\n    "},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"})"}
]
